js.go

```diff
@@ -1222,10 +1222,15 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync,
 				}
 				return nil, fmt.Errorf("nats: %s", cinfo.Error.Description)
 			}
-		} else if consumer == _EMPTY_ {
-			// Update our consumer name here which is filled in when we create the consumer.
+		} else {
+			// Since we created the JS consumer internally, mark that we should
+			// delete it on Unsubscribe().
```

Comment:
I think the intention to delete an underlying consumer may need to be more explicit.

Reply:
I have strong opinion that the library should try to delete (and ok if it does not since if really it is ephemeral the server will delete it anyway) *only* if it called AddConsumer() and got OK in the subscribe() call.

Reply:
Again think of app that wants to use durable just to survive restart but once its done it wants the consumer deleted too. If the app crashes, the second version will never delete the durable.

Reply:
Then app should call AddConsumer() -> subscribe -> process all -> DeleteConsumer. Again, we are trying to do too much in the library and assume we know what the user's end goal is.

Reply:
(but as you suggested on Slack, if we make it an option, then I am fine with that)

Reply:
Let's do that, we never delete the underlying durable by default, you can DeleteConsumer or we can add a SubOpt to js.Subscribe() that instructs the system to do so.